DB設計

R-carrot opened the pull request:
# what 

DBのER図

# why

フリマアプリに必要なDBをわかりやすく整理するため

R-carrot committed:
初期設定

R-carrot committed:
DBの設計情報

chobi555 commented:
レビュー依頼ありがとうございます！
お手数ですが、トレロの要件にございます通り、「ER図」のご提出をお願いできればと存じます。
PRにER図を添付し、再度レビュー依頼をお願いいたします。

わからないことございましたら、コネクトにて質問していただくようお願いいたします。

R-carrot committed:
productsをitemsに変更

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| Column                 | Type                | Options                 |
+|------------------------|---------------------|-------------------------|
+| email                  | string              | null: false             |
+| password               | string              | null: false             |
```
```
| password | string | null: false |
```
こちらのカラムはdeviseのgemを使用しての実装になるため、deviseが予め用意している
encrypted_passwordというカラムを用意してあげましょう。

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
-* Ruby version
+| Column                 | Type                | Options                 |
+|------------------------|---------------------|-------------------------|
+| email                  | string              | null: false             |
```
```
| email | string | null: false |
```
こちらemailのカラムには一意性制約をかけましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| last_name              | string              | null: false             |
+| first_name_katakana    | string              | null: false             |
+| last_name_katakana     | string              | null: false             |
+| birth_year             | integer             | null: false             |
+| birth_month            | integer             | null: false             |
+| birth_day              | integer             | null: false             |
```
誕生日に関してですが、年、月、日を1つで指定して取得する方法があります。
「rails date型」と調べてみると、ヒントが見つかりますので、一度調べてみましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
-* System dependencies
 
-* Configuration
+### Association
```
こちらbuyersテーブルとのアソシエーションも考えてみましょう。

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| ------------------ | ------------------- | ----------------------- |
+| title              | string              | null: false             |
+| explain            | text                | null: false             |
+| category           | integer             | null: false             |
+| state              | integer             | null: false             |
+| delivery_fee       | integer             | null: false             |
+| owner_region       | integer             | null: false             |
+| delivery_date      | integer             | null: false             |
```
商品の状態, 配送料の負担, 発送元の地域,  発送までの日数, カテゴリーのカラムはactive_hashで実装の予定のため、integer型にしましょう！

R-carrot replied:
この指摘だけよくわかりませんでした。
integer型になっていると思うのですが、違うやり方があるということでしょうか。

yutamasumoto replied:
失礼しました。
こちらactive_hashで実装の予定のため、 _id を付けて実装しましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+
+
+
+## comments table
```
commentsテーブルに関しては必須機能では必要ない機能ですので削除しておきましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| Colum              | Type                | Options                 |
+| ------------------ | ------------------- | ----------------------- |
+| user               | references          | foreign_key             |
+| items              | references          | foreign_key             |
```
```
| items | references | foreign_key |
```
カラム名は単数形にしておきましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| items              | references          | foreign_key             |
+
+
+### Association
```
userテーブルとのアソシエーションも考えて見ましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+
+| Colum              | Type                | Options                 |
+| ------------------ | ------------------- | ----------------------- |
+| card_number        | string              | null: false             |
+| effective_year     | string              | null: false             |
+| effective_month    | string              | null: false             |
+| card_security_code | string              | null: false             |
```
```
| card_number        | string              | null: false             |
| effective_year     | string              | null: false             |
| effective_month    | string              | null: false             |
| card_security_code | string              | null: false             |
```
こちらカード情報に関してはセキュリティの観点からデーターベースで保存するのは適しておりません。削除しておきましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| effective_month    | string              | null: false             |
+| card_security_code | string              | null: false             |
+| postal_code        | string              | null: false             |
+| prefecture         | integer             | null: false             |
```
```
| prefecture | integer | null: false |
```
こちらに関しましてはactive_hashで実装の予定のため、カラム名に _id を付けましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+
+
+
+## delivery_info table
```
buyersテーブルの外部キーをカラムに追加しましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,102 @@
+### Association
+
+- belongs_to :buyer
+- belongs_to :user
+- belongs_to :item
```
```
- belongs_to :user
- belongs_to :item
```
こちらのアソシエーションは不要となるので削除しておきましょう！

yutamasumoto reviewed (commented):
レビュー依頼ありがとうございます！
12点コメントしましたので、ご確認をお願いいたします！

yutamasumoto reviewed (commented):
コメント頂きありがとうございます。
こちらコメントさせて頂きましたのでご確認の程お願いいたします。

R-carrot committed:
指摘された箇所の修正

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,97 @@
 
-* Database creation
+- has_many :items
+- has_many :comments
```
```
- has_many :comments
```
こちらcommentsテーブルを実装から外していただいていると思いますのでこちらの記述も削除しましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,97 @@
-* Database creation
+- has_many :items
+- has_many :comments
+- has_one  :buyer
```
```
-has_one :buyer
```
こちらuserは複数の商品を購入することができますので1対多の形にしましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,97 @@
+
+### Association 
+
+- has_many    :comments
```
```
- has_many :comments
```
こちらcommentsテーブルを実装から外していただいていると思いますのでこちらの記述も削除しましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,97 @@
+| item               | references          | foreign_key: true       |
+ -->
+
+### Association
+
+- belongs_to :user
+- belongs_to :items
```
こちらcommentsテーブルのアソシエーションだと思いますので、こちらも削除しておきましょう！

yutamasumoto commented on README.md:
```diff
@@ -1,24 +1,97 @@
+| ------------------ | ------------------- | ----------------------- |
+| user               | references          | foreign_key: true       |
+| item               | references          | foreign_key: true       |
+| delivery_info      | references          | foreign_key: true       |
```
```
| delivery_info | references | foreign_key: true |
```
こちらbuyersテーブルにdelivery_infosテーブルの外部キーを追加するのではなく、delivery_infosテーブルにbuyersテーブルの外部キーを追加しましょう。

yutamasumoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
5点追加でコメントいたしましたので、ご確認お願いします！

R-carrot committed:
指摘箇所の修正２

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,97 @@
+| last_name              | string              | null: false              |
+| first_name_katakana    | string              | null: false              |
+| last_name_katakana     | string              | null: false              |
+| birth_day              | datetime            | null: false              |
```
こちらはdate型で実装しましょう！
(時刻は使用しないためです)

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,97 @@
+| Colum              | Type                | Options                 |
+| ------------------ | ------------------- | ----------------------- |
+| postal_code        | string              | null: false             |
+| prefecture_id      | integer             | null: false             |
```
都道府県のモデルは商品テーブルと同じものを使い回すことができるため、こちらも「owner_region_id」というカラム名にしてあげましょう！
(またはどちらも「prefecture_id」としてもよいですね！)

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
2点追加でコメントいたしましたので、ご確認お願いします！
あと少しでございます！頑張りましょう！

R-carrot committed:
指摘箇所の修正３

datetime型をdate型に修正（時刻を使用しないため）
owner_region_idをprefecture_idへ変更（delivery_info tableと同じにして再利用可能にした）

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
最初の関門を越えましたね！
このブランチをマージし、masterブランチでプルを行いましょう！
反映されていることを確認し、引き続き実装頑張っていきましょう！
